### PR TITLE
add rnaseq indep specimen lists for release

### PR DESCRIPTION
**Pull request by jharenza**
#### What scientific question is your analysis addressing?
Non-cancer group list (same as all-cohort) of independent RNA-Seq specimens is required for the v11 release, in order to run fusion filtering.

#### What was your approach?
- Add one script to randomly select rnaseq specimens across cohorts, with this run requiring `histologies-base.tsv`
- Updated the run scripts as well.

#### What GitHub issue does your pull request address?
NA

#### Is there anything that you want to discuss further?
No

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/independent-samples/run-independent-samples.sh [hunk 1]
```diff
@@ -14,6 +15,11 @@ RUN_FOR_SUBTYPING=${OPENPBTA_BASE_SUBTYPING:-0}
 # Set the working directory to the directory of this file
 cd "$(dirname "${BASH_SOURCE[0]}")"
 
+if [[ "$RUN_FOR_SUBTYPING" -eq "1" ]]
+then
+# Run rna-seq only
+Rscript 04-generate-independent-specimens-rnaseq-release.R
```
Comment (logstar) [inline analyses/independent-samples/run-independent-samples.sh]:
```suggestion
  # Run rna-seq only
  Rscript 04-generate-independent-specimens-rnaseq-release.R
```

Indent for easier to read.

Could you also indent after `else` and before `fi`?

File: scripts/run-for-subtyping.sh [hunk 2]
```diff
@@ -13,6 +13,10 @@ OPENPBTA_BASE_SUBTYPING=1
 
 ## Step 1. Generate summary files needed for subtyping
 
+# Create the independent sample list using the *FULL* histology file
```
Comment (logstar) [inline scripts/run-for-subtyping.sh]:
```suggestion
# Create the independent sample list using the *base* histology file, i.e., histologies-base.tsv
```

Not sure if I understand correctly. I assume full histology file is `histologies.tsv`, which is generated after subtyping.

File: analyses/independent-samples/04-generate-independent-specimens-rnaseq-release.R [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+# Filter to only samples from tumors from RNA-Seq minus GTEX
+# Note that there are some samples with unknown composition, but these will be ignored for now.
+rnaseq_samples <- histology_df %>%
+  dplyr::filter(sample_type == "Tumor", 
+                composition != "Derived Cell Line", 
+                experimental_strategy == "RNA-Seq")
+
+print(nrow(rnaseq_samples))
+# generate WGS independent samples for all cohorts
```
Comment (logstar) [inline analyses/independent-samples/04-generate-independent-specimens-rnaseq-release.R]:
```suggestion
# generate release RNA-seq independent samples for all cohorts
```

File: analyses/independent-samples/run-independent-samples.sh (hunk 1 above)
Comment (jharenza) [reply analyses/independent-samples/run-independent-samples.sh]: done

**Comment by jharenza** [comment]
>Thank you for preparing this PR. I wonder if the PR should be merged/compared to the v11-release branch rather than the dev branch.

Yes, was thinking about this, and it does require v11 for the base file. I chose `dev` because of the amount of stacking of PRs we are doing right now, so I think putting this into `dev `would be better than putting this in the middle of a stack somewhere... 😬

**Comment by jharenza** [comment]
@logstar ready for re-review

File: analyses/independent-samples/README.md [hunk 4]
```diff
@@ -21,14 +23,18 @@ Komal Rathi, Run Jin, Yuanchao Zhang
...
 ├── results
 │   ├── independent-specimens.rnaseq.primary-plus.eachcohort.tsv
 │   ├── independent-specimens.rnaseq.primary-plus.tsv
+│   ├── independent-specimens.rnaseq.primary-plus-release.tsv
 │   ├── independent-specimens.rnaseq.primary.eachcohort.tsv
 │   ├── independent-specimens.rnaseq.primary.tsv
+│   ├── independent-specimens.rnaseq.primary-release.tsv
 │   ├── independent-specimens.rnaseq.relapse.eachcohort.tsv
 │   ├── independent-specimens.rnaseq.relapse.tsv
+│   ├── independent-specimens.rnaseq.relapse-release.tsv
```
Comment (runjin326) [inline analyses/independent-samples/README.md]: Just a thought - would it be better to use `pre-release` to indicate they are not ready for release yet?

**Review by runjin326 — APPROVED** [review]
Everything makes sense and checked all logic - they all work!

File: analyses/independent-samples/README.md (hunk 4 above)
Comment (jharenza) [reply analyses/independent-samples/README.md]: good idea!

**Review by logstar — COMMENTED** [review]
@jharenza - Thank you for updating the PR. I agree it is more straightforward to compare this PR with the `dev` branch. All `v11-release` based branches can be further updated, after this PR is merged.

The code/comment/documentation changes look good to me. Rerun in docker reproduced the results.

I was wondering if README.md needs to be update to describe the two modes of `independnet-samples` runs. Following is a draft:

````markdown
## Generating sample lists

There are two modes for generating independent sample lists:

- Release mode: Release independent sample lists are generated using the `histologies-base.tsv` file, and these lists are used to run modules specific to release (eg: fusion filtering).

  ```bash
  OPENPBTA_BASE_SUBTYPING=1 ./analyses/independent-samples/run-independent-samples.sh
  ```

- Analysis mode: Independent sample lists are generated per cohort and per 
cohort and cancer group, which requirs `histolologies.tsv`.

  ```bash
  OPENPBTA_BASE_SUBTYPING=0 ./analyses/independent-samples/run-independent-samples.sh
  ```

````

**Comment by jharenza** [comment]
@logstar updated the README

**Pull request merged**